announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -694,11 +703,42 @@ else if (view.equals(VIEW_MODE_PUBLIC))
 							{
 								messages = getMessagesPublic(site, channel, null, true, state, portlet);
 							}
+							else if (VIEW_MODE_BYROLE.equals(view))
+							{
+								messages = getMessagesByRoles(site, channel, null, true, state, portlet);
+							}
 						}
 						else
 						{
 							messages = getMessages(channel, null, true, state, portlet);
 						}
+						List<AnnouncementWrapper> finalMessages = new ArrayList<>();
+						for (AnnouncementWrapper message : messages) {
+							site = SiteService.getSite(EntityManager.newReference(message.getReference()).getContext());
+							String selectedRoles = (String) message.getProperties().get(SELECTED_ROLES_PROPERTY);
+							boolean addAnnounMessage = false;
+							if (selectedRoles == null) {
```

Comment:
Shall we check the superUser here ?

Reply:
Yes

Reply:
Let's be consistent with the entity provider.

String ownerId = a.getAnnouncementHeader().getFrom().getId();
String userId = sessionManager.getCurrentSessionUserId();
boolean isOwner = ownerId.equals(userId);

if (isOwner || securityService.isSuperUser() || selectedRoles == null) {
} else {
}